### PR TITLE
broker/boot_config: use new config file interfaces

### PR DESCRIPTION
**Pull request by garlick**
This PR just updates the "config file bootstrap" code in the broker to use the new `flux_conf_t` interface for accessing the config object.

### CODE REVIEW

**Comment by garlick** [comment]
restarted one builder that hit a valgrind timeout:
```
== ERROR SUMMARY: 0 errors from 0 contexts (suppressed: 0 from 0)
Alarm clock
not ok 1 - valgrind reports no new errors on 2 broker run
```

File: src/broker/boot_config.h [hunk 1]
```diff
@@ -30,18 +32,23 @@
  *
  *   # if commented out, instance size is the size of tbon-endpoints.
  *   size = 3
+ *
+ * Caveat: the tbon-endpoints array entries specify a ZMQ endpoint used
+ * in both "bind" and "connect" API calls.  ZMQ "bind" endpoints accept
+ * an IP address or an interface name.  ZMQ "connect" endpoints accept
+ * an IP address or hostname.  Therefore only IP's -- not hostnames, and not
```
Comment (grondo) [inline src/broker/boot_config.h]: nit: incorrect use of apostrophe in `IP's`

**Comment by garlick** [comment]
Doh!  My Achilles heel.  Forced a push.

**Comment by garlick** [comment]
Thanks!

**Pull request merged**